### DISCUSSION TITLE
Login with username or email

### DISCUSSION THREAD

**Discussion by FrenchGithubUser** [lexik/LexikJWTAuthenticationBundle#1025] [Q&A]
My user entity has a username and an email.
Is it possible to login (obtain a token) by providing either the username or the email (letting the user choose the option, like on many websites).
Couldn't find anything about that on the web

**Comment by fd6130** [comment]
Sure you can, just create your own custom login api and create the token there.

https://symfony.com/bundles/LexikJWTAuthenticationBundle/current/7-manual-token-creation.html

**Reply by FrenchGithubUser** [reply]
there is no parameter that we could add in the `lexik_jwt_authentication.yaml` file (instead of creating our own custom controller) ?

**Reply by fd6130** [reply]
Refer from https://github.com/lexik/LexikJWTAuthenticationBundle/blob/2.x/Resources/doc/1-configuration-reference.rst, I don't think it have such parameter there.
